pkg/cmd/run.go

```diff
@@ -278,6 +298,14 @@ func fetchSecrets(localConfig models.ScopedOptions, enableFallback bool, fallbac
 				}
 			}
 		}
+
+		if enableCache {
```

Comment:
Not sure if this is needed, but you are doing it in other places:
`if enableCache && metadataPath != "" {`

Reply:
Great catch